tools/lkl/cptofs.c

```diff
@@ -675,6 +675,26 @@ int main(int argc, char **argv)
 			break;
 	}
 
+	/* FIXME: Remouting as readonly seems to be required to make sure the filessytem is cleanly umounted. Otherwise the journal will be still dirty... */
```

Comment:
Looks good as a workaround, but I would like to investigate the root cause. Do you have a way to reproduce this?

Reply:
```
$ truncate -s 10G /tmp/foo.img
$ mkfs.ext4 /tmp/foo.img         
$ ./tools/lkl/cptofs -p -o 1000 -g 1000 -t ext4 -i /tmp/foo.img ./tools/lkl/tests  / 
```

Now if you try to mount the image read-only, you can see hat it fails because of the unclean journal

```
$ mount -o ro /tmp/foo.img /mnt
```

Reply:
@tavip did you managed to reproduce the issue? This would block us updating lkl in NixOS, where we use it for building disk images without the need for KVM.

Reply:
I think I found the root cause but it is turning out more difficult to fix than I expected. Lets merge this for now. Thanks for the ping.